include/stronk/unit_v2.h

```diff
@@ -22,43 +24,67 @@ concept unit_value_like = ::twig::stronk_like<T> && requires { typename T::unit_
 // Implementations
 
 // Tag should either be the unit itself, or a dimensions_like type defining the dimensions of the unit.
-template<typename Tag, template<typename StronkT> typename... SkillTs>
+template<typename Tag, typename ScaleT, template<typename StronkT> typename... SkillTs>
 struct unit
 {
     using dimensions_t = std::conditional_t<dimensions_like<Tag>, Tag, twig::create_dimensions_t<dimension<Tag, 1>>>;
+    using scale_t = ScaleT;
 
     unit() = delete;  // Do not construct this type
 
+    // we recreate ratio to make sure we have the gcd version of the scale.
+    template<typename OtherScaleT>
+    using scaled_t = unit<Tag, std::ratio<OtherScaleT::num, OtherScaleT::den>, SkillTs...>;
+
     template<typename UnderlyingT>
     struct value : ::twig::stronk<value<UnderlyingT>, UnderlyingT, SkillTs...>
     {
         using unit_t = unit;
         using base_t = ::twig::stronk<value<UnderlyingT>, UnderlyingT, SkillTs...>;
         using base_t::base_t;
 
-        // can convert to same unit with different underlying type
-        template<typename T>
-        constexpr explicit operator value<T>() const
+        // can convert to same unit with different underlying type, or different scale
+        template<typename NewUnderlyingT>
+            requires(!std::same_as<NewUnderlyingT, UnderlyingT>)
+        constexpr explicit operator value<NewUnderlyingT>() const
+        {
+            return value<NewUnderlyingT> {static_cast<NewUnderlyingT>(this->val())};
+        }
+
+        template<typename NewScaleT, typename NewUnderlyingT = UnderlyingT>
+        constexpr auto to() const -> scaled_t<NewScaleT>::template value<NewUnderlyingT>
         {
-            return value<T> {static_cast<T>(this->val())};
+            if constexpr (std::same_as<NewScaleT, ScaleT>) {
+                return static_cast<value<NewUnderlyingT>>(*this);
+            } else {
+                using converter = std::ratio_divide<ScaleT, NewScaleT>;
+                using result_value_t = scaled_t<NewScaleT>::template value<NewUnderlyingT>;
+                return result_value_t {static_cast<NewUnderlyingT>(this->val()) * converter::num / converter::den};
```

Comment:
Would be nice if we could document some limits of the conversion. I suppose we would loose precision very fast.

Reply:
Maybe you can propose what you think would be nice here?

Reply:
I'm not really sure. It just feels like a pretty way of converting small numbers to 0 because of changing scale.
Or big numbers to overflow.

Maybe it is fine?

Reply:
I can rename the function to convert? But for sure its probable that you will get a lossy conversion - I think its okay that its up to the user to do that.

Reply:
How do you feel about the final version I made: No implicit underlying conversion anymore